assessments/projects/knative/self-assessment.md

```diff
@@ -0,0 +1,577 @@
+
+2. Role-Based Access Control (RBAC): Knative aims to nicely integrate with Kubernetes RBAC at the control-plane layer. This helps ensure that only authorized users and services can interact with and modify KNative deployments.
+
+3. Container Image Security: Knative ships all of the manifests with content-digests of the image, as well as SBOMs stored in the container registry and keep the dependencies up to date. Knative Serving by default resolves image tags to digests on deployment of a revision.This avoids a number of failure modes when the resolution of an image tag changes after initial deployment.
+
+### Non-Goals
+- Long-Term Data Storage - KNative may not intend to provide a solution for long-term data storage. Users might be directed to leverage other storage solutions or databases that are better suited for persistent data.
+- Direct Management of Kubernetes Resources - KNative might not aim to expose all the intricacies of Kubernetes resource management directly to end-users. The abstraction provided by KNative is designed to shield users from low-level Kubernetes operations.
+- Fine-grained Network Security Policy - Knative may not aim to offer fine-grained network security policy management at the application level. Users are encouraged to leverage Kubernetes Network Policies or other networking solutions for such requirements.
+
+---
+ 
+## Self-assessment Use
+This self-assessment is created by a team of cybersecurity students at NYU to perform an external analysis of the project's security. It is not intended to provide a security audit of Knative, or function as an independent assessment or attestation of Knative's security health.
+
+This document serves to provide Knative users with an initial understanding of Knative's security, where to find existing security documentation, Knative plans for security, and general overview of Knative security practices, both for development of Knative as well as security of Knative.
+
+This document provides the CNCF TAG-Security with an initial understanding of Knative to assist in a joint-assessment, necessary for projects under incubation. Taken together, this document and the joint-assessment serve as a cornerstone for if and when Knative seeks graduation and is preparing for a security audit.
+
+---
+
+## Security functions and features
+| Component | Applicability | Description of Importance |
+| --------- | ------------- | ------------------------- |
+| Code Signature Verification | Security Relevant | Knative signs their releases with cosign to verify binaries, ensuring integrity and authenticity for users. |
+| Istio | Security Relevant | Istio is an open platform-independent service mesh that provides traffic management, policy enforcement, and telemetry collection. Although Istio is not required for Knative to operate, it can be used for networking operations like traffic rules, authorization policies, and encryption. |
```

Comment:
This section should include functionalities within knative that enable key security functions. This is not meant to include third party tools or functionalities (such as Istio or security gaurd, i'm unfamiliar with the latter). Please update this section to reflect Knatives security functions